<?xml version="1.0"?>
<launch>

    <param name="use_sim_time" value="true"/>

    <node pkg="stefmap_ros" type="stefmap_node_online.py" name="stefmap_node" output="screen">
    	<param name="grid_size" value="1"/>  
		  <param name="x_min" value="-25" />
    	<param name="x_max" value="25" />
    	<param name="y_min" value="-25" />
    	<param name="y_max" value="25" />
    	<param name="interval_time" value="10" />
    	<param name="num_bins" value="8" />
    	<param name="frame_id" value="map" />
    	<param name="people_detections_topic" value="/people_tracker/pose_array" />
    	<param name="coverage_laser_topic" value="/velodyne_scan" />
    	<param name="coverage_time_update" value="5" />
    </node>

	<node pkg="fremenarray" type="fremenarray" name="fremenarray" output="screen">
    </node>

    
    <node pkg="tf" type="static_transform_publisher" name="static_transform_publisher" args="0 0 0 0 0 0 /map /odom  1000" output="log"/>

    <node pkg="velodyne_pointcloud" type="cloud_node" name="cloud_node" output="log">
        <param name="calibration" value="$(find velodyne_pointcloud)/params/VLP16db.yaml"/>
    </node>

  <node pkg="pointcloud_to_laserscan" type="pointcloud_to_laserscan_node" name="pointcloud_to_laserscan" output="log">
      <remap from="cloud_in" to="/velodyne_points"/>
      <remap from="scan" to="/velodyne_scan"/>
      <rosparam>
          #target_frame: camera_link # Leave disabled to output scan in pointcloud frame
          transform_tolerance: 0.01
          min_height: 0.2 #change according the heigh where the real sense is
          max_height: 1.5
          angle_min: -3.14 # -M_PI
          angle_max: 3.14 # M_PI
          angle_increment: 0.0043 # M_PI/360.0
          scan_time: 0.1
          range_min: 0.1
          range_max: 30
          use_inf: true
          # Concurrency level, affects number of pointclouds queued for processing and number of threads used 0 : Detect number of cores 1 
          # : Single threaded
          # 2->inf : Parallelism level
          #concurrency_level: 1
      </rosparam>
  </node>


</launch>